packages/adp-tooling/src/preview/routes-handler.ts

```diff
@@ -0,0 +1,198 @@
+                return {
+                    fragmentName: key
+                };
+            });
+
+            res.status(HttpStatusCodes.OK)
+                .contentType('application/json')
+                .send({
+                    fragments,
+                    filteredFragments: filteredFragmentsWithoutChangeFile,
+                    message: `${fragments.length} fragments found in the project workspace.`
+                });
+            this.logger.debug(`Read fragments ${JSON.stringify(fragments)}`);
+            this.logger.debug(
+                `Read fragments without change file ${JSON.stringify(filteredFragmentsWithoutChangeFile)}`
+            );
+        } catch (e) {
+            this.logger.error(e.message);
+            res.status(HttpStatusCodes.INTERNAL_SERVER_ERROR).send({ message: e.message });
+            next(e);
+        }
+    };
+
+    /**
+     * Handler for writing a fragment file to the workspace
+     *
+     * @param req Request
+     * @param res Response
+     * @param next Next Function
+     */
+    public handleWriteFragment = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            const data = req.body as WriteFragmentBody;
+
+            const fragmentName = sanitize(data.fragmentName);
+
+            const projectPath = process.cwd();
+
+            if (fragmentName) {
+                const fullPath = path.join(projectPath, FolderNames.Webapp, FolderNames.Changes, FolderNames.Fragments);
+                const filePath = path.join(fullPath, `${fragmentName}.fragment.xml`);
+
+                if (!fs.existsSync(fullPath)) {
+                    fs.mkdirSync(fullPath);
+                }
+
+                if (fs.existsSync(filePath)) {
+                    res.status(HttpStatusCodes.CONFLICT).send(`Fragment with name "${fragmentName}" already exists`);
+                    return;
+                }
+
+                // Copy the template XML Fragment to the project's workspace
+                const fragmentTemplatePath = path.join(__dirname, '../templates', TemplateFileName.Fragment);
+                fs.copyFileSync(fragmentTemplatePath, filePath);
+
+                const message = 'XML Fragment created';
+                res.status(HttpStatusCodes.CREATED).send(message);
+            } else {
+                res.send(HttpStatusCodes.BAD_REQUEST).send('Fragment Name was not provided!');
+            }
+        } catch (e) {
+            const sanitizedMsg = sanitize(e.message);
+            this.logger.error(sanitizedMsg);
+            res.status(HttpStatusCodes.INTERNAL_SERVER_ERROR).send(sanitizedMsg);
+            next(e);
+        }
+    };
+
+    /**
+     * Handler for reading the manifest.appdescr_variant contents
+     *
+     * @param req Request
+     * @param res Response
+     * @param next Next Function
+     */
+    public handleReadAppDescrVariant = async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
You don't need this additional handler. All files in the `webapp` folder are always hosted at `/` e.g. the app descriptor variant is at `/manifest.appdescr_variant`

Reply:
@nikmace I think there is a misunderstanding. Not even what you have in 8abb0ba is required. You need no code at all. The content of the `webapp` folder is exposed by `ui5 serve` by default.